vm-monitor/src/main/java/com/epam/pipeline/vmmonitor/service/vm/VMMonitor.java

```diff
@@ -102,7 +106,7 @@ private void checkVmState(final VirtualMachine vm) {
                 checkMatchingNodes(nodes, vm);
             } else {
                 log.debug("No matching nodes were found for VM {} {}.", vm.getInstanceId(), vm.getCloudProvider());
-                if (!matchingRunExists(vm)) {
+                if (!matchingRunExists(vm) && !poolIdExists(vm)) {
```

Comment:
Pool label exists only on kubernetes nodes, so we shall check it in `checkLabels` method, which operates kubernetes `Node` instead of AWS `VirtualMachine`